### PR TITLE
XSS sniff: support implode() and array_map()

### LINKED ISSUE #442: implode() use causes a false positive
The following code triggers an `Expected a sanitizing function...` warning, despite the output of `implode()` being safe:

```php
$items = array( 'foo', 'bar' );
echo implode( '<br>', esc_html( $items ) );
```

### PR DESCRIPTION
**Pull request by JDGrimes**
Because `implode()` is mostly useless without `array_map()`.

Fixes #442

### CODE REVIEW

File: WordPress/Sniffs/XSS/EscapeOutputSniff.php [hunk 1]
```diff
@@ -274,19 +274,41 @@ public function process( PHP_CodeSniffer_File $phpcsFile, $stackPtr )
...
+				if ( $function_opener ) {
+
+					if ( 'array_map' === $functionName ) {
+
+						// Get the first parameter (name of function being used on the array).
+						$mapped_function = $this->phpcsFile->findNext(
+							PHP_CodeSniffer_Tokens::$emptyTokens
+							, $function_opener + 1
+							, $tokens[ $function_opener ]['parenthesis_closer']
+							, true
```
Comment (westonruter) [inline WordPress/Sniffs/XSS/EscapeOutputSniff.php]: @JDGrimes I think we previously talked about this comma-first style not being the style for WPCS?

Comment (GaryJones) [inline WordPress/Sniffs/XSS/EscapeOutputSniff.php]: That's correct Weston.

Comment (JDGrimes) [inline WordPress/Sniffs/XSS/EscapeOutputSniff.php]: Oops! I actually copied this from another part of the code (which I must have written :smile:), but I'll correct it.

Comment (JDGrimes) [inline WordPress/Sniffs/XSS/EscapeOutputSniff.php]: Fixed in 6b786e9.

File: WordPress/Sniffs/XSS/EscapeOutputSniff.php [hunk 2]
```diff
@@ -274,19 +274,41 @@ public function process( PHP_CodeSniffer_File $phpcsFile, $stackPtr )
 			// Now check that next token is a function call.
 			if ( T_STRING === $this->tokens[ $i ]['code'] ) {
 
+				$ptr = $i;
+
 				$functionName = $this->tokens[ $i ]['content'];
+
+				$function_opener = $this->phpcsFile->findNext( array( T_OPEN_PARENTHESIS ), $i + 1, null, null, null, true );
```
Comment (westonruter) [inline WordPress/Sniffs/XSS/EscapeOutputSniff.php]: Gotta love positional arguments :smile:

File: WordPress/Sniff.php [hunk 3]
```diff
@@ -304,6 +304,7 @@
 	 */
 	public static $formattingFunctions = array(
 		'ent2ncr' => true,
+		'implode' => true,
```
Comment (westonruter) [inline WordPress/Sniff.php]: I think `join` should be added here too, as it is an alias of `implode`.

**Comment by westonruter** [comment]
:clap: excellent work as always, @JDGrimes

**Pull request merged**